### PR TITLE
[RI-597]Refactor and reuse leapfrog preflight checks for incrementals

### PR DESCRIPTION
**Pull request by rexredinger**
The goal is to selectively include preflight checks for
incremental and all future upgrades in a maintainable way.

### CODE REVIEW

**Comment by rexredinger** [comment]
recheck_trusty_aio_swift_mitaka_to_newton_leap

File: playbooks/preflight_checks/common.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        - enable_lxc_fs_check | default(true) | bool
+      with_items: "{{ lxc_fs_check.stdout_lines | default(['LXC FS check was skipped.']) }}"
+
+- name: Check for apt_pinned_packages in openstack_deploy
+  hosts: localhost
+  gather_facts: false
+  user: root
+  tasks:
+    - name: Check for apt_pinned_packages in overrides
+      command: "grep 'apt_pinned_packages' user_rpco_variables_overrides.yml user_osa_variables_overrides.yml"
```
Comment (BjoernT) [inline playbooks/preflight_checks/common.yml]: Search across all `user*.yml` files if possible

**Comment by rexredinger** [comment]
This PR is now waiting on var changes from #328

**Comment by rexredinger** [comment]
From the trusty_aio_swift_liberty_to_newton_leap:
> 2019-02-20T10:45:09-0600 :stderr: DEPRECATION: --allow-all-external has been deprecated and will be removed in the future. Due to changes in the repository protocol, it no longer has any effect.
> 2019-02-20T10:45:09-0600 fatal: unable to access 'https://github.com/fireteam/virtualenv-tools/': Could not resolve host: github.com
> 2019-02-20T10:45:09-0600 Command "git clone -q https://github.com/fireteam/virtualenv-tools /tmp/src/virtualenv-tools" failed with error code 128 in None

Seems a bit odd.  That command works locally.  But also lots if pip install failures on many containers

> 2019-02-20T10:18:33-0600   Could not find a version that satisfies the requirement pip==9.0.0 

as well as:

> 2019-02-20T10:04:39-0600 failed: [aio1_galera_container-340f1495] => {"failed": true}
> 2019-02-20T10:04:39-0600 msg: Mysql fact collection failed: "ERROR 2003 (HY000): Can't > connect to MySQL server on '127.0.0.1' (111 "Connection refused")".

Ugh

**Comment by rexredinger** [comment]
recheck_trusty_aio_swift_mitaka_to_newton_leap

**Comment by antonym** [comment]
recheck_trusty_aio_swift_mitaka_to_newton_leap

**Comment by rexredinger** [comment]
recheck_trusty_aio_swift_liberty_to_newton_leap
recheck_trusty_aio_swift_mitaka_to_newton_leap

**Pull request closed without merging**